src/database-event-rule.ts

```diff
@@ -0,0 +1,183 @@
+import { IEventBus, IRuleTarget, Rule, Schedule } from 'aws-cdk-lib/aws-events';
+// @ts-ignore
+// eslint-disable-next-line no-duplicate-imports
+import type { RuleProps } from 'aws-cdk-lib/aws-events';
+import { IDatabaseInstance } from 'aws-cdk-lib/aws-rds';
+import { Construct } from 'constructs';
+
+/**
+ * Category of the database instance event.
+ *
+ * Useful for filtering down to specific event types.
+ *
+ * @ref {@link https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Events.Messages.html#USER_Events.Messages.instance}
+ */
+export enum DatabaseEventCategory {
+  /** Database availability event: shutdown, restart, etc. */
+  AVAILABILITY = 'availability',
+  /** Database backup event: started backup, finished backup, etc. */
+  BACKUP = 'backup',
+  /**
+   * Database configuration change event: updated parameter group, modified
+   * instance class, reset credentials, etc.
+   */
+  CONFIGURATION_CHANGE = 'configuration change',
+  /** Database creation event: instance created. */
+  CREATION = 'creation',
+  /** Database deletion event: instance deleted. */
+  DELETION = 'deletion',
+  /** Database failover event: failover started, failover complete, etc. */
+  FAILOVER = 'failover',
+  /** Database failure event: instance failure, etc. */
+  FAILURE = 'failure',
+  /** Database low storage event: storage exhausted, etc. */
+  LOW_STORAGE = 'low storage',
+  /** Database maintenance event: instance patched, version upgrade, etc. */
+  MAINTENANCE = 'maintenance',
+  /** Database maintenance failure event: update of Oracle time zone failed. */
+  MAINTENANCE_FAILURE = 'maintenance, failure',
+  /** Database maintenance notification event: time zone update, etc. */
+  MAINTENANCE_NOTIFICATION = 'maintenance, notification',
+  /**
+   * Database notification event: patching delayed, operator issued
+   * notification, exceeding best practices, etc. */
+  NOTIFICATION = 'notification',
+  /** Database read replica event: replication started, stopped, etc. */
+  READ_REPLICA = 'read replica',
+  /** Database recovery event: recovery started, complete. etc. */
+  RECOVERY = 'recovery',
+  /** Database restoration event: restored instance. */
+  RESTORATION = 'restoration',
+  /** Database security event: decrypting HSM password to update instance. */
+  SECURITY = 'security',
+  /** Database security patching event: system update available. */
+  SECURITY_PATCHING = 'security patching',
+}
+
+/**
+ * Properties for defining a database EventBridge rule.
+ *
+ * @see {@link RuleProps}
+ */
+export interface DatabaseEventRuleProps {
```

Comment:
Did you mean to extend `RuleProps` here? Since you're extending the class `Rule`, you probably should.

Thinking about it more, you'll run into constraint problems if you don't satisfy `RuleProps`, since you're extending `Rule` and passing the props to it.

You might want to consider extending just `Construct` and creating the `Rule` inside your construct.

Reply:
Originally, I extended `RuleProps` omitting `eventPattern` but `jsii` does not support the `Omit` utility type. I considered extending `Construct` but since `DatabaseEventRule` has an "is a" relationship to `Rule` I though it better to extend `Rule`. The advantage being that `DatabaseEventRule` automatically gets all the functionality associated with `Rule`, and can be treated as a `Rule`, without any further implementation. If you think it is better to extend `Construct` then I can switch to that.

The props of `Rule` are satisfied since `DatabaseEventRuleProps` is duck typed to match the interface of `RuleProp`.

Ref: https://docs.aws.amazon.com/prescriptive-guidance/latest/best-practices-cdk-typescript-iac/constructs-best-practices.html#create-l2-construct